src/main/scala/org/scanet/core/CoreOp.scala

```diff
@@ -65,19 +65,43 @@ import scala.language.higherKinds
   @op("<<", alias = true)
   def dependsOn[A: TensorType](op: F[A], dep: F[_]): F[A]
 
+  /** Asserts that given condition (constructed from current tensor by specified fn)
+   * is true or fail graph execution (and prints current value).
+   *
+   * {{{
+   * val a = Tensor.vector(1, 2).const
+   * val b = Tensor.vector(3, 4).const
+   * val c = Tensor.vector(4, 6).const
+   *
+   * (a plus b).assert(_ === c).eval should be(Tensor.vector(4, 6))
+   * }}}
+   *
+   * @param f function to build assertion condition from current op
+   * @return current op for chaining
+   */
+  def assert[A: TensorType](op: F[A], f: F[A] => F[Boolean]): F[A]
+
   /** Cast elements of given tensor form type A into B.
    * Returns given input if A is already equal to B.
    *
    * @return casted output
    */
   def cast[A: TensorType, B: TensorType](op: F[A]): F[B]
+
+  /** Return current op as leaf operation - means it doesnt produce output
+   * and can be evaluated in graph only if added as dependant operation.
+   *
+   * @see dependsOn
+   * @return current leaf node
+   */
+  def asLeaf[A: TensorType](op: F[A]): F[Nothing]
```

Comment:
Not sure that `asLeaf` is the best name, basically in this expression `c = a + b` both `a` and `b` are leaves. Maybe we can call it something like `asUnit`, `asVoid`, `noop`, ..?

Reply:
renamed into `asVoid`